packages/rule-tester/src/RuleTester.ts

```diff
@@ -929,130 +941,143 @@ export class RuleTester extends TestFramework {
-                    actualSuggestion.desc,
-                    expectedDesc,
-                    `${suggestionPrefix} desc should be "${expectedDesc}" but got "${actualSuggestion.desc}" instead.`,
-                  );
-                }
-
-                if (hasOwnProperty(expectedSuggestion, 'messageId')) {
-                  assert.ok(
-                    ruleHasMetaMessages,
-                    `${suggestionPrefix} Test can not use 'messageId' if rule under test doesn't define 'meta.messages'.`,
-                  );
-                  assert.ok(
-                    hasOwnProperty(
-                      rule.meta.messages,
-                      expectedSuggestion.messageId,
-                    ),
-                    `${suggestionPrefix} Test has invalid messageId '${expectedSuggestion.messageId}', the rule under test allows only one of ${friendlyIDList}.`,
-                  );
-                  assert.strictEqual(
-                    actualSuggestion.messageId,
-                    expectedSuggestion.messageId,
-                    `${suggestionPrefix} messageId should be '${expectedSuggestion.messageId}' but got '${actualSuggestion.messageId}' instead.`,
-                  );
-
-                  const unsubstitutedPlaceholders =
-                    getUnsubstitutedMessagePlaceholders(
-                      actualSuggestion.desc,
-                      rule.meta.messages[expectedSuggestion.messageId],
-                      expectedSuggestion.data,
-                    );
 
-                  assert.ok(
-                    unsubstitutedPlaceholders.length === 0,
-                    `The message of the suggestion has ${unsubstitutedPlaceholders.length > 1 ? `unsubstituted placeholders: ${unsubstitutedPlaceholders.map(name => `'${name}'`).join(', ')}` : `an unsubstituted placeholder '${unsubstitutedPlaceholders[0]}'`}. Please provide the missing ${unsubstitutedPlaceholders.length > 1 ? 'values' : 'value'} via the 'data' property for the suggestion in the context.report() call.`,
-                  );
-
-                  if (hasOwnProperty(expectedSuggestion, 'data')) {
-                    const unformattedMetaMessage =
-                      rule.meta.messages[expectedSuggestion.messageId];
-                    const rehydratedDesc = interpolate(
-                      unformattedMetaMessage,
-                      expectedSuggestion.data,
+                error.suggestions!.forEach(
+                  (expectedSuggestion: SuggestionOutput<MessageIds>, index) => {
+                    assert.ok(
+                      typeof expectedSuggestion === 'object' &&
+                        expectedSuggestion !== null,
+                      "Test suggestion in 'suggestions' array must be an object.",
+                    );
+                    // @ts-expect-error -- we purposely don't define `desc` on our types as the current standard is `messageId`
+                    const expectedDesc = expectedSuggestion.desc as string;
```

Comment:
These two lines used to be in the `if (hasOwnProperty(expectedSuggestion, 'desc')) { ...`, but they have now been moved out of that block. I'm note sure if this change is related to this PR. WDYT?